pkg/fieldpath/paved.go

```diff
@@ -204,6 +204,8 @@ func expandWildcards(data any, segments Segments) ([]Segments, error) { //nolint
 					}
 					res = append(res, r...)
 				}
+			case nil:
+				return nil, errNotFound{errors.Errorf("%s: value of the field is nil", segments[:i])}
```

Comment:
nit:
```suggestion
				return nil, errNotFound{errors.Errorf("value of the field %q is nil", segments[:i])}
```

Reply:
Here, we differentiate between the cases where an unexpected type is encountered versus the specified path is not found and we hit the new added `case nil` here.